x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/insights_tab_csp.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo } from 'react';
+import { EuiButtonGroup, EuiSpacer } from '@elastic/eui';
+import type { EuiButtonGroupOptionProps } from '@elastic/eui/src/components/button/button_group/button_group';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useExpandableFlyoutState } from '@kbn/expandable-flyout';
+import { MisconfigurationFindingsDetailsTable } from './misconfiguration_findings_details_table';
+
+const insightsButtons: EuiButtonGroupOptionProps[] = [
+  {
+    id: 'misconfigurationTabId',
+    label: (
+      <FormattedMessage
+        id="xpack.securitySolution.flyout.left.insights.misconfigurationButtonLabel"
+        defaultMessage="Misconfiguration"
+      />
+    ),
+    'data-test-subj': 'misconfigurationTabDataTestId',
+  },
+];
+
+/**
+ * Insights view displayed in the document details expandable flyout left section
+ */
+export const InsightsTabCsp = memo(({ name }: { name: string }) => {
+  const panels = useExpandableFlyoutState();
+  const activeInsightsId = panels.left?.path?.subTab ?? 'misconfigurationTabId';
+
+  return (
+    <>
+      <EuiButtonGroup
+        color="primary"
+        legend={i18n.translate('xpack.securitySolution.flyout.left.insights.optionsButtonGroups', {
+          defaultMessage: 'Insights options',
+        })}
+        options={insightsButtons}
+        idSelected={activeInsightsId}
+        onChange={() => {}}
+        buttonSize="compressed"
+        isFullWidth
+        data-test-subj={'insightButtonGroupsTestId'}
+      />
+      <EuiSpacer size="xl" />
+      <MisconfigurationFindingsDetailsTable fieldName={'host.name'} queryName={name} />
```

Comment:
having the `fieldName` fixed to `host.name` here might prevent this component from being reusable in the future, I would suggest including the `fieldName` in the props and pass it down from the `getInsightsInputTab` function.

Reply:
was planning to just do that on my next ticket (do this with user.name column) as shown in this [branch](https://github.com/elastic/kibana/commit/ea08349bfd452c64715b4ade5c4d55f6d77c5ed6) that i made to see how much stuff i need to change 
but now that u mentioned it, might as well change it now